R/enum.R

```diff
@@ -0,0 +1,35 @@
+
+#' Create an enumeration to be used instead of arbitraty values in code.
+#'
+#' @param inputList
+#'
+#' @return the Enum created
+#' @export
+#'
+enum <- function(inputList) {
+  myEnum <- as.list(inputList)
+  enumNames <- names(myEnum)
+  if (is.null(enumNames)) {
+    names(myEnum) <- myEnum
+  } else if ("" %in% enumNames) {
+    stop("The inputList has some but not all names assigned. They must be all assigned or none assigned")
+  }
+  return(myEnum)
+}
+
+ContainerType <- enum(c(

```

Comment:
@PavelBal This is a quick try to implement enum like struct based on some code I found
It works great but I have one issue. The enum I create, such as `ContainerType` need to be in the same file as the enum function otherwise it is not found 

Any idea why?

This probably due to some namespace issue. In short here, I want to declare a global variable to the package called `ContainerType` that everyone can access by doing for example `ContainerType$Organ`

Let's think about it a bit